assets/js/components/higherorder/withData.js

```diff
@@ -141,6 +147,23 @@ const withData = (
 		return false;
 	}
 ) => {
+	/**
+	 * Map of data requests by context.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @type {Object.<string, Object[]>}
+	 */
+	const dataRequestsByContext = selectData.reduce(
+		( acc, dataRequest ) => {
+			castArray( dataRequest.context ).forEach( ( context ) => {
+				acc[ context ] = acc[ context ] || [];
+				acc[ context ].push( dataRequest );
+			} );
+			return acc;
+		},
+		{}
+	);
 	// ...and returns another component...
 	class NewComponent extends Component {
```

Comment:
The changes here will likely be easier to review with whitespace changes hidden
![image](https://user-images.githubusercontent.com/1621608/105350696-35576580-5bf4-11eb-8326-a161d6eff0e7.png)